### PR TITLE
Remove drag & drop dropping items from containers

### PR DESCRIPTION
**Pull request by SlamBamActionman**
## About the PR

This PR makes dragging items out of the container UI no longer drop the item on the ground.

## Why / Balance

The drag & drop is a nice feature to have, but it bypasses a few key systems in a way that wouldn't be intuitive. This includes forensics, glue/lube and dual wielding.

Originally the PR hindered players from moving items entirely in their bag while their hands were full, however after discussion this was decided to be QoL to justify it. Dropping, on the other hand, was seen as more of an obscure mechanic and since it changes the item state it's less desireable.

## Technical details

This simply removes the drop functionality and all accompanying code.

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- tweak: It's no longer possible to drag an item out of a container's UI to drop it.

### CODE REVIEW

**Comment by bruhmogus** [comment]
i don't see why this would make sense, irl you can just... move it with your arm

**Comment by packmore** [comment]
how is this any more intuitive than just being allowed to move things in your inventory whenever? this is a goofy space game, no reason to be going anti-QoL just for realism.

**Comment by metalgearsloth** [comment]
> how is this any more intuitive

Moving entities around your inventory already requires a hand, making storage require it is more consistent and I would argue more intuitive due to this.

**Comment by SlamBamActionman** [comment]
> i don't see why this would make sense, irl you can just... move it with your arm

You have to make concessions for stuff like glue and forensics to work, or something like #25253. If you're moving an object around, you're interacting by touching it, and if you're touching it you should require a free hand.

**Comment by Erisfiregamer1** [comment]
> Moving entities around your inventory already requires a hand, making storage require it is more consistent and I would argue more intuitive due to this.

while I agree it'd be slightly more intuitive, this is just annoying as a casual player. in a real life situation, as bruhmogus said, I'd just use my arm to move it if my hands were full.

**Comment by packmore** [comment]
> Moving entities around your inventory already requires a hand, making storage require it is more consistent and I would argue more intuitive due to this.

i guess i'm looking at it from the perspective of "intuitive for a video game" and not "intuitive for someone who has never played a video game". like, i wouldn't automatically assume that my item not moving in my inventory is because of my filled hand, i'd assume i'm lagging or that there's a bug. i might figure it out quickly, but i think it would just be more frustrating and be an overall negative payoff.

**Comment by metalgearsloth** [comment]
> i wouldn't automatically assume that my item not moving in my inventory is because of my filled hand

Does this mean you should be able to unequip inventory items even if your hands are full?

**Comment by packmore** [comment]
> Does this mean you should be able to unequip inventory items even if your hands are full?

this is a false equivalence imo. equipped items are inserted into your hands when unequipped, and there is no other way to take the equipped items off (as far as i'm aware). also, the equip UI has no reason to be organized and it doesn't really affect QoL in any way. being able to move things around while holding an item then quickly store the item makes sense and doesn't even require you to put that item down IRL - why should it in game when it only adds needless frustration? not to mention, being able to organize your items in a container UI is a pretty obvious game abstraction that doesn't really make sense to force into being more realistic.

**Comment by UbaserB** [comment]
I actually never knew it was possible to move items with something in your hand, I always picked it up and placed it into the inventory... I've been playing the game wrong lol (but honestly this pr will change nothing for me so its fine)

**Comment by Centronias** [comment]
To set a bit of context, presently with full hands, you can
* Click and drag to rearrange items in storage
* Click and drag to remove items from storage (they're dropped on the floor under your sprite)

, and you _cannot_
* Click and drag to pick up items from the world, either to inventory (ie. pockets) or storage; or to equip
  * because click and drag doesn't do anything to items in the world -- the click tries to pick up, but fails because full hands
* Click and drag to equip items from storage, either to equipment slots, pockets or hands
  * This is perhaps the most inconsistent one for somebody who doesn't know how the UI works because...
  * It just drops the item on the ground. The clicked-and-dragged ghost UI thingy doesn't interact with the inventory/pockets/hands UI at all, and it acts like you're just dropping the item in the world

So, perhaps the broader question @metalgearsloth is asking here is whether or not the inventory/hands/pockets should work more like storage. That is, specifically, if it should have the same sort of click-and-drag capability.
And then the flipside would be if storage should work more like the inventory/hands/pockets and require an open hand to make any position changes.

File: Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs [hunk 1]
```diff
@@ -1522,6 +1532,27 @@ private bool ValidateInput(EntitySessionEventArgs args,
         return true;
     }
 
+    /// <summary>
+    /// Check if the entity interacting with the container is doing so with an empty hand.
+    /// </summary>
+    private bool ValidateEmptyHand(EntitySessionEventArgs args, [NotNullWhen(true)] out Hand? emptyHand)
```
Comment (Centronias) [inline Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs:1538]: IMO a name like `TryGetEmptyHand` (to parallel `TryComp`) fits better with the `out var` semantics.
But at the same time I have next to no experience with `out var`s, so I dunno.

**Comment by Graded5** [comment]
i dont really like that you cant move items if you have a empty hand but its not your active hand

**Comment by SlamBamActionman** [comment]
Based on a Maintainer Discussion, this PR has been adjusted to still allow moving items in your inventory when your hands are full; however dropping items from your inventory or a container with drag & drop has been removed completely. The reasoning for this is that it bypasses some checks, and that it's a fairly unintuitive and hidden mechanic to begin with.

**Comment by cfif126** [comment]
THIS SUCKS ASS!!!!!

**Comment by thebadman4662** [comment]
Borgs no longer can get bodybags from boxes because doctors are allergic to preventing rot. :mistake:
Mice griefers no longer can get shit out to drag spill and otherwise abuse bugs as much. :godo:

**Comment by archee1** [comment]
Personally I disagree with the core reasons for this PR's balancing, I am a very avid user of this feature and can attest to having never seen any of the points made happen in real gameplay, while yes they are odd interactions such as dropping glued items, this rarely occurs in gameplay nor for experienced players is ever considered as an option in the first place.

As for forensics I doubt its a problem to add to the code to put glove fibres/finger prints onto items interacted this way, akin to how bags are apparently touched when opened despite never entering your hands (even if said item has an 'opened' sprite in the case of pizza boxes/honk meal boxes/toolboxes). likewise transmitting lube-dropping onto something bag-dropped or preventing glued items from being bag-dropped would also be better solutions to prevent these points.

To say this feature is obscure however is a fair point to make (Though a large portion of this game is niche knowledge, we arent hailing the removal of drag-emptying jugs into a chemmaster for this reason), there-in a setting to enable/disable the overall functionality would also be a more acceptable alternative to removing it, while also dealing with its 'obscureness'.

Overall this change came out of left field to me and has heavily upset my micro with inventory interactions and I would very much be pleased to see it undone.
Managing bags will always be a chore, this was an incredibly useful QOL that has vanished; For the same reason that needing a free hand to re-arrange your bag was scrapped, speaks volumes regarding my stance.

**Pull request merged**